src/python/grpcio/grpc/_channel.py

```diff
@@ -142,7 +142,7 @@ def _handle_event(event, state, response_deserializer):
         response = _common.deserialize(
             serialized_response, response_deserializer)
         if response is None:
-          details = b'Exception deserializing response!'
+          details = 'Exception deserializing response!'
```

Comment:
In Python3.4, byte strings are particularly annoying to manage.  I believe some of the ```details``` you were creating used ```.format()``` which is not supported for byte strings.  Pretty much all the other nice functions on ```str``` don't work either.

For that reason, I relaxed the API to also accept ascii encodable ```str``` types.  This is similar to what the library was doing before with the Python2 ```unicode``` strings.  It would accept them, and then utf-8 encode them before passing them to the c-core.  I felt that utf-8 encoding was too lenient because there are multiple canonical encodings for non-ascii characters, and so I restricted it to ascii characters.